File: blind-75/arrays-and-hashing/contains-duplicate/typescript/contains-duplicate.ts
export function containsDuplicate(nums: number[]): boolean {
  const mapNums = new Map<number, number>();

  for (const num of nums) {
    if (!mapNums.has(num)) {
      mapNums.set(num, 1);
    } else {
      return true;
    }
  }

  return false;
}

File: blind-75/arrays-and-hashing/contains-duplicate/java/ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
  public boolean containsDuplicate(int[] nums) {
    Set<Integer> setNums = new HashSet<>();

    for (int num : nums) {
      if (!setNums.add(num)) {
        return true;
      }
    }

    return false;
  }
}
